File: 1503-last-moment-before-all-ants-fall-out-of-a-plank/1503-last-moment-before-all-ants-fall-out-of-a-plank.java
class Solution {
    public int getLastMoment(int n, int[] left, int[] right) {
        if (right.length != 0 && left.length != 0) {
            int a1 = Integer.MIN_VALUE;
            for (int i : left) {
                a1 = Math.max(a1, i);
            }
            
            int a2 = Integer.MAX_VALUE;
            for (int i : right) {
                a2 = Math.min(a2, i);
            }
            
            return Math.max(n - a2, a1);
        } else if (right.length == 0) {
            int a1 = Integer.MIN_VALUE;
            for (int i : left) {
                a1 = Math.max(a1, i);
            }
            return a1;
        } else {
            int a2 = Integer.MAX_VALUE;
            for (int i : right) {
                a2 = Math.min(a2, i);
            }
            return n - a2;
        }
    }
}

File: 1503-last-moment-before-all-ants-fall-out-of-a-plank/1503-last-moment-before-all-ants-fall-out-of-a-plank.py
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        if right and left:
            a1=max(left)
            a2=min(right)
            return max(n-a2,a1)
        elif not right:
            a1=max(left)
            return a1
        else:
            a2=min(right)
            return n-a2
        